regression/run.t

```diff
@@ -0,0 +1,2847 @@
+
+  $ build-irmin >> /dev/null 2>&1
+
+  $ merl-an error-regression -s 1 --data=test-data 2> /dev/null
```

Comment:
```suggestion
  $ merl-an error-regression --data=test-data 2> /dev/null
```
One sample per file is very little. `merl-an`'s default is 30. It'd be interesting to see how long the run takes with 30 samples per file.

(Side-note: this has reminded me that `merl-an` should distribute the samples better throughout the project by making the number of samples per file depend on the size of the file/AST. I've just opened an issue on `merl-an` to remember: https://github.com/pitag-ha/merl-an/issues/28)

Reply:
Also, having -s 30 means, that I have wait so long to get results and slows my dev process.

Reply:
> Also, I think that having more than 2,5k tests isn't small.

I agree that even with one sample per file, this isn't small. In fact, that's the reason why I've suggested to make it a bit more realistic: That way, we could find out if this approach scales or if we should consider choosing a different approach.

> You can test bigger runs on different branches.

That sounds good. Have you done that already?

If we merge it like this without investigating more, we just need to be aware that we might end up removing this and implementing a different approach later. I have a tendency to first investigate things and then merge them, but I know that both you and Thibaut like doing it the other way around and that's ok as well.

Reply:
I have tested this on my machine:
```
-- Results --
machine: Laptop with Intel i7 11 gen
sample size: 30 (default)

time:
real 203m8.524s
user 0m35.541s
sys 0m17.571
```
The results between CI and my dev env are comparable, because we use single thread.

> If we merge it like this without investigating more, we just need to be aware that we might end up removing this and implementing a different approach later.

I'm surprised, that we are considering using different approach at this point. I thought that we already discussed this CI and agreed on its implementation. I don't see why we would need to remove it, instead of simply adding some tweaks.

> I have a tendency to first investigate things and then merge them, but I know that both you and Thibaut like doing it the other way around and that's ok as well.

It's hard for me to agree with this remark. I think I have investigated a lot before creating this PR. What I'm suggesting, is that we already provide a value with this PR, so we can merge it and then improve according to specific needs.

Reply:
Oh, I'm sorry, I wasn't aware you had already tested it with sample-size 30. How big is the test file when you do that?

If you've already investigated enough on your side and are convinced that this approach is the right way to go, I'm happy to believe you and stop the thinking/discussion around the approach itself to focus only on details/concrete improvements. Is that what you were saying?

Reply:
```
artifact size:
$ ls -l --block-size=M
total 11M
-rwxr-xr-x 1 opam opam 8M Jun 25 23:43 commands.json
-rwxr-xr-x 1 opam opam 1M Jun 25 23:43 logs.json
-rwxr-xr-x 1 opam opam 2M Jun 25 23:43 results.json
```
It was measured without cmds, so it could be bigger (let's say x2).

I think that this approach looks promising. I can't guarantee that it will scale perfectly to all requirements. I think it's something that we can ask about @voodoos too.